R-package/src/install.libs.R

```diff
@@ -42,17 +42,23 @@ if (!use_precompile) {
   setwd(build_dir)
   
   # Prepare installation steps
-  cmake_cmd <- "cmake "
-  build_cmd <- "make _lightgbm -j4"
+  cmake_base <- "cmake "
+  build_cmd <- "make _lightgbm -j"
   lib_folder <- file.path(R_PACKAGE_SOURCE, "src", fsep = "/")
   
   # Check if Windows installation (for gcc vs Visual Studio)
   if (WINDOWS) {
     if (use_mingw) {
-      cmake_cmd <- paste0(cmake_cmd, " -G \"MinGW Makefiles\" ")
-      build_cmd <- "mingw32-make.exe _lightgbm -j4"
+      cmake_cmd <- paste0(cmake_base, " -G \"MinGW Makefiles\" ")
+      build_cmd <- "mingw32-make.exe _lightgbm -j"
+      system(paste0(cmake_cmd, " ..")) # Must build twice for Windows due sh.exe in Rtools
     } else {
-      cmake_cmd <- paste0(cmake_cmd, " -DCMAKE_GENERATOR_PLATFORM=x64 ")
+      cmake_cmd <- paste0(cmake_base, " -DCMAKE_GENERATOR_PLATFORM=x64 ")
+      tryVS <- system(paste0(cmake_cmd, " .."))
+      if (tryVS == 1) {
+        cmake_cmd <- paste0(cmake_base, " -G \"MinGW Makefiles\" ") # Switch to MinGW on failure, try build once
```

Comment:
Do you know where is the build folder in all the folders? I can't seem to get rid of it.

Objective would be:
- try VS build
- if VS build fails, try MinGW build but delete build folder first (I don't know where is the build folder in all the folders)